htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -568,6 +568,13 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 			// Update detail stock for batch product
 			if (!$error && !empty($conf->productbatch->enabled) && $product->hasbatch() && !$skip_batch)
 			{
+				// check unicity for serial numbered equipments ( different for lots managed products)
+				if ( $product->status_batch == 2 && $qty > 0 && $this->getBatchCount($fk_product, $batch) > 0 )
+				{
+					$error++;
+					$this->errors[] = $langs->trans("SerialNumberAlreadyInUse", $batch, $product->ref);
+				}
+
 				if ($id_product_batch > 0)
```

Comment:
I suggest to add a test here 
`if (!$error) {`
here to execute the creation of batch record only if previous test on unicity of serial is ok

Reply:
Hello @eldy , thanks for the advices; the new pr is done :)